src/robotFactory.js

```diff
@@ -1,17 +1,77 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
 }
 
-function FlyingRobot() {
-  // implement
-}
+BaseRobot.prototype = {
+  goForward(step = 1) {
+    this.coords.y += step;
+  },
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  },
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  },
 
-function DeliveryDrone() {
-  // implement
+  goRight(step = 1) {
+    this.coords.x += step;
+  },
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion},`
+      + ` Weight: ${this.weight}`;
+  },
+};
+
+function FlyingRobot(name, weight, coords, chipVersion) {
+  BaseRobot.call(this, name, weight, coords, chipVersion);
+
+  this.coords = {
+    z: 0,
+  };
```

Comment:
```suggestion
  this.coords.z = 0;
```
думаю так було б краще